lib/galaxy/tools/__init__.py

```diff
@@ -2431,6 +2441,7 @@ def exec_before_job(self, app, inp_data, out_data, param_dict=None):
 class ExpressionTool(Tool):
     requires_js_runtime = True
     tool_type = 'expression'
+    tool_type_local = True
```

Comment:
This doesn't have to be local ? It just needs galaxy + node, right ?

Reply:
In your framework tests it seems to write stuff to the job directory that Pulsar doesn't know about it - so it will fail with Pulsar. I think we could change this if we fixed Pulsar - I would imagine approaching this like switching this variable when we update the client staging logic to account for these files.

Reply:
If you add a `TODO` here I'm happy to merge this

Reply:
https://github.com/galaxyproject/pulsar/issues/263